ipa-core/src/helpers/mod.rs

```diff
@@ -405,23 +406,25 @@ impl TryFrom<[Role; 3]> for RoleAssignment {
 /// Combination of helper role and step that uniquely identifies a single channel of communication
 /// between two helpers.
 #[derive(Clone, Eq, PartialEq, Hash, Ord, PartialOrd)]
-pub struct ChannelId {
-    pub role: Role,
+pub struct ChannelId<I: transport::Identity> {
+    pub peer: I,
```

Comment:
This name: `peer` confuses me. Why did you change this from `role`? What does it mean here? And can you add a comment?

Reply:
the reason I had to rename it is that this struct is now used to communicate between shards too and there is no notion of helper role for such communications. `ChannelId<ShardIndex>` would have `role: ShardIndex` field which looks even more confusing, hence a neutral `peer` that can be anything that wants to communicate with this instance